### PR TITLE
Update lib part one - interface

### PR DESCRIPTION
**Pull request by afischerdev**
At #441 we talked about splitting the big update into smaller ones to have a better overview.
This is the first part with 'only' interface changes. It has no new results, routines for using it will follow.

### CODE REVIEW

**Review by zod — CHANGES_REQUESTED** [review]
I know that I might be rather picky but for me it's the wrong order. I think first the functionality has to be added to the lower layers and afterwards exposed at the interface. But that's just my preference, we can also work our way down.

The PR title is misleading, this PR doesn't change anything regarding the lib but the http and android service interface. Please use a more meaningful title.

File: brouter-routing-app/src/main/aidl/btools/routingapp/IBRouterService.aidl [hunk 1]
```diff
@@ -28,9 +31,9 @@ interface IBRouterService {
     //  "exportWaypoints" = 1 to export them (optional, default is no export)
     //  "pois"            = lon,lat,name|... (optional)
     //  "extraParams"     = Bundle key=value list for a profile setup (like "profile:")
-    //  "timode"          = turnInstructionMode [0=none, 1=auto-choose, 2=locus-style, 3=osmand-style, 4=comment-style, 5=gpsies-style, 6=orux-style] default 0
+    //  "timode"          = turnInstructionMode [0=none, 1=auto-choose, 2=locus-style, 3=osmand-style, 4=comment-style, 5=gpsies-style, 6=orux-style, 7=locus-new-style, 8=cruiser-stylem, 9=brouter-intern] default 0
     //  "heading"         = angle (optional to give a route a start direction)
-    //  "direction"       = (deprecated) angle
+    //  "direction"       = angle (optional, used on recalculation, only used by Locus)
```
Comment (polyscias) [inline brouter-routing-app/src/main/aidl/btools/routingapp/IBRouterService.aidl:36]: Would be good to indicate how it is used by Locus

File: brouter-routing-app/src/main/java/btools/routingapp/BRouterWorker.java [hunk 2]
```diff
@@ -57,7 +57,7 @@ public String getTrackFromParams(Bundle params) {
       if ("osmand".equalsIgnoreCase(tiFormat)) {
         rc.turnInstructionMode = 3;
       } else if ("locus".equalsIgnoreCase(tiFormat)) {
-        rc.turnInstructionMode = 2;
+        rc.turnInstructionMode = 7;
       }
```
Comment (polyscias) [inline brouter-routing-app/src/main/java/btools/routingapp/BRouterWorker.java:61]: Are we not missing OsmAnd here?

Comment (afischerdev) [reply brouter-routing-app/src/main/java/btools/routingapp/BRouterWorker.java:61]: How do you mean that? OsmAnd is still present. Only timode for locus is switched to new standard output.

File: brouter-routing-app/src/main/aidl/btools/routingapp/IBRouterService.aidl (hunk 1 above)
Comment (afischerdev) [reply brouter-routing-app/src/main/aidl/btools/routingapp/IBRouterService.aidl:36]: I have no view inside Locus. But I can see Locus sends a 'direction' tag via the interface.

File: brouter-routing-app/src/main/java/btools/routingapp/BRouterWorker.java (hunk 2 above)
Comment (polyscias) [reply brouter-routing-app/src/main/java/btools/routingapp/BRouterWorker.java:61]: You are completely right, not sure what I saw.

File: brouter-routing-app/src/main/aidl/btools/routingapp/IBRouterService.aidl (hunk 1 above)
Comment (afischerdev) [reply brouter-routing-app/src/main/aidl/btools/routingapp/IBRouterService.aidl:36]: It is not ignored by brouter. 
It is stored at [BRouterWorker](https://github.com/abrensch/brouter/blob/eba0b18689d03d0ff3611a8896c4b593e45d8d15/brouter-routing-app/src/main/java/btools/routingapp/BRouterWorker.java#L67)
And used in this situation at [RoutingEngine](https://github.com/abrensch/brouter/blob/eba0b18689d03d0ff3611a8896c4b593e45d8d15/brouter-core/src/main/java/btools/router/RoutingEngine.java#L615) when an older calculation is found.

Comment (polyscias) [reply brouter-routing-app/src/main/aidl/btools/routingapp/IBRouterService.aidl:36]: Okay, I see it is used as start direction largely following "heading", how about making that clear in the description:

angle (optional, used like "heading" on a recalculation request by Locus as start direction)

Comment (afischerdev) [reply brouter-routing-app/src/main/aidl/btools/routingapp/IBRouterService.aidl]: made an update in your way

Comment (polyscias) [reply brouter-routing-app/src/main/aidl/btools/routingapp/IBRouterService.aidl]: Thanks!

**Review by zod — APPROVED** [review]

**Pull request merged**